<launch>
  <arg name="traffic_short_topic_name" default="/apollo/sensor/camera/traffic/image_short" />
  <arg name="traffic_long_topic_name" default="/apollo/sensor/camera/traffic/image_long" />
  <arg name="frame_rate" default="30"/>

  <group ns="camera_long">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_traffic_long" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
        <param name="video_device" value="/dev/video0" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
        <param name="camera_timeout" value="2000" />
        <param name="error_code" value="11" />
        <remap from="/camera_long/image_raw0" to="$(arg traffic_long_topic_name)" />
        <remap from="/sensor/camera/traffic/camera_info" to="$(arg traffic_long_topic_name)/camera_info" />
    </node>
  </group>
</launch>
